actors/miner/tests/util.rs

```diff
@@ -2279,9 +2284,341 @@ fn fixed_hasher(offset: ChainEpoch) -> Box<dyn Fn(&[u8]) -> [u8; 32]> {
+    state: &State,
+    store: &BS,
+    allocated_sectors: &BTreeSet<u64>,
+    acc: &MessageAccumulator,
+) {
+    let quant = state.quant_spec_every_deadline(policy);
+
+    // invert pre-commit clean up queue into a lookup by sector number
+    let mut cleanup_epochs: BTreeMap<u64, ChainEpoch> = BTreeMap::new();
+    match BitFieldQueue::new(store, &state.pre_committed_sectors_cleanup, quant) {
+        Ok(queue) => {
+            let ret = queue.amt.for_each(|epoch, expiration_bitfield| {
+                let epoch = epoch as ChainEpoch;
+                let quantized = quant.quantize_up(epoch);
+                acc.require(
+                    quantized == epoch,
+                    &format!("pre-commit expiration {epoch} is not quantized"),
+                );
+
+                expiration_bitfield.iter().for_each(|sector_number| {
+                    cleanup_epochs.insert(sector_number, epoch);
+                });
+                Ok(())
+            });
+            acc.require_no_error(ret, "error iterating pre-commit clean-up queue");
+        }
+        Err(e) => {
+            acc.add(&format!("error loading pre-commit clean-up queue: {e}"));
+        }
+    };
+
+    let mut precommit_total = BigInt::zero();
+
+    let precommited_sectors =
+        Map::<_, SectorPreCommitOnChainInfo>::load(&state.pre_committed_sectors, store);
+
+    match precommited_sectors {
+        Ok(precommited_sectors) => {
+            let ret = precommited_sectors.for_each(|key, precommit| {
+                let sector_number = match parse_uint_key(&key) {
+                    Ok(sector_number) => sector_number,
+                    Err(e) => {
+                        acc.add(&format!("error parsing pre-commit key as uint: {e}"));
+                        return Ok(());
+                    }
+                };
+
+                acc.require(
+                    allocated_sectors.contains(&sector_number),
+                    &format!("pre-commited sector number has not been allocated {sector_number}"),
+                );
+
+                acc.require(
+                    cleanup_epochs.contains_key(&sector_number),
+                    &format!("no clean-up epoch for pre-commit at {}", precommit.pre_commit_epoch),
+                );
+                precommit_total += &precommit.pre_commit_deposit;
+                Ok(())
+            });
+            acc.require_no_error(ret, "error iterating pre-commited sectors");
+        }
+        Err(e) => {
+            acc.add(&format!("error loading precommited_sectors: {e}"));
+        }
+    };
+
+    acc.require(state.pre_commit_deposits == precommit_total,&format!("sum of pre-commit deposits {precommit_total} does not equal recorded pre-commit deposit {}", state.pre_commit_deposits));
+}
+
 #[allow(dead_code)]
-pub fn check_state_invariants(_rt: &MockRuntime) {
-    // TODO check state invariants
+pub fn check_state_invariants(rt: &MockRuntime) -> (StateSummary, MessageAccumulator) {
```

Comment:
For the convenience of integration tests (and out of package callers) we'll want to use the same signature as the go implementation passing in all of the flattened components and avoiding the mock runtime.

Reply:
done